Putting it all together section (#120)
jaybee84/ml-in-rd: content/06.multiple-approaches-required.md

#### Multiple approaches are required 

We have described multiple approaches for maximizing the success of machine learning applications in rare disease research throughout. 
In practice, it is rarely sufficient to use one of these techniques in isolation. 

> In this section, we will discuss two recent works – one supervised and one unsupervised – in the rare disease domain that draw on multiple concepts covered here. Feature-representation-transfer (and therefore representation learning and prior data) and regularization underlie both approaches. 

Reviewer: ```suggestion
In this section, we will discuss two recent works – one supervised and one unsupervised – in the rare disease domain that draw on multiple concepts covered here.
Feature-representation-transfer (and therefore representation learning and prior data) and regularization underlie both approaches. 
```

As discussed in the transfer learning section, feature-representation-transfer is an approach wherein a model learns a representation of some collection of data and those features are applied or transferred to a target domain.
This can be a powerful tool for rare diseases if the model used requires more samples than available samples that assay the disease of interest, but we have a collection of data that is related.
Whether our research goal is biological discovery or prediction, we want the representation that a model learns to be _useful_.
In addition to reducing model complexity as discussed earlier, regularization can aid in useful representations by putting some constraints on what is learned by a model.

In a method termed DeepProfile, Dincer et al. leveraged publicly available acute myeloid leukemia (AML) gene expression data to improve the prediction of _in vitro_ drug responses [@doi:10.1101/278739].
The authors trained a variational autoencoder (an unsupervised neural network that learns a series of representations from data), or VAE, on AML data that had been collected over time without the desired phenotypic information (drug response).

> VAEs are distinct from other autoencoders in that the distribution of the encodings are regularized such that they are close to a normal distribution, which can help avoid overfitting and learn a representation that is more than just an identity mapping [TODO: CITATIONS / Take another pass at this sentence].

Reviewer: It definitely regularizes by adding the additional constraint. However, the extent of that constraint (kinda heavily discussed in the beta-VAE literature since that focuses on weighting the two constraints) and how that produces this outcome is still a bit unclear.  There are a few results in Greg's BioBombe paper ( https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02021-3 ) that also hint at this VAE constraint as being helpful in this way.

Author: Yep, this was on my mind! I wasn't sure how to thread this needle in a sentence or two. It may be helpful to just come out and say that some of these things are unclear.

#### Multiple approaches are required 

We have described multiple approaches for maximizing the success of machine learning applications in rare disease research throughout. 
In practice, it is rarely sufficient to use one of these techniques in isolation. 
In this section, we will discuss two recent works – one supervised and one unsupervised – in the rare disease domain that draw on multiple concepts covered here.

> Feature-representation-transfer (and therefore representation learning and prior data) and regularization underlie both approaches. 

Reviewer 2: ```suggestion
These two approaches combine dimension reduction techniques like representation learning with prior knowledge and use transfer learning techniques to perform feature-representation-transfer from large related datasets to smaller datasets.
Furthermore, in these examples feature-representation-transfer utilizes regularization to identify disease-relevant features that are useful for constructing simple but stable models.
```

Reviewer 2: As I re-read the lines following this, I think you also hit the exact points that I was thinking of (just in different words). I am still adding these lines here, but you can decide to keep them or not :)

> As discussed in the transfer learning section, feature-representation-transfer is an approach wherein a model learns a representation of some collection of data and those features are applied or transferred to a target domain.

Reviewer 2: ```suggestion
As discussed in the previous section, feature-representation-transfer is an approach wherein a model learns a representation of a collection of data, and those features are then applied or transferred to a target domain.
```

> This can be a powerful tool for rare diseases if the model used requires more samples than available samples that assay the disease of interest, but we have a collection of data that is related.

Reviewer 2: ```suggestion
This is a powerful tool where the model can utilize related datasets in the absence of available rare disease samples.
```

(Here "unseen" refers to the fact that these diseases were not in the training set.) 
PLIER is a matrix factorization approach that takes prior knowledge in the form of gene sets or pathways and gene expression data as input. 
PLIER includes constraints (regularization) such that some latent variables learned by the model will align with input gene sets and ideally latent variables will only be associated with a low number of related gene sets [@doi:10.1038/s41592-019-0456-1], which make it suitable for biological discovery or description of rare disease data.
MultiPLIER allows us to use one model to describe multiple datasets instead of reconciling output from multiple models, which is highly beneficial when identifying commonalities among disease manifestations or affected tissues is a research goal. 

> Taken together, DeepProfile [@doi:10.1101/278739] and MultiPLIER [@doi:10.1016/j.cels.2019.04.003] suggest a combination of the techniques discussed throughout can be capitalized on for rare disease research.

Reviewer 2: ```suggestion
Taken together, DeepProfile [@doi:10.1101/278739] and MultiPLIER [@doi:10.1016/j.cels.2019.04.003] suggest a combination of the techniques discussed throughout this article can be capitalized on for rare disease research.
```

> We may have few samples from our disease of interest with the required phenotypic labels, but we can leverage existing collections of data and knowledge if we select the models with the _right attributes_.

Reviewer 2: ```suggestion
In cases where we have few samples from our disease of interest with the required phenotypic labels, we can leverage existing collections of data and knowledge if we select the models with the _right attributes_.
```
